src/engine/strat_engine/thinpool/thinpool.rs

```diff
@@ -553,18 +552,16 @@ impl ThinPool {
             }
         }
 
-        let filesystems = self
-            .filesystems
-            .borrow_mut()
-            .iter_mut()
-            .map(|(_, _, fs)| fs.check())
-            .collect::<StratisResult<Vec<_>>>()?;
-
-        for fs_status in filesystems {
+        for (name, uuid, fs) in self.filesystems.iter_mut() {
+            let (fs_status, save_mdv) = fs.check()?;
+            if save_mdv && self.mdv.save_fs(name, uuid, &fs).is_err() {
+                error!("Failed to save MDV for : {}", uuid);
+            }
```

Comment:
@mulkieran any suggestions for doing this in a more idiomatic way?  I couldn't get it working with the closure.